Bug: v1.3.0 amazon-ebs unexpected EOF

Packer crashes when using v1.3.0 even though template is validated successfully. Same template works on v1.2.5. I could not locate the crash.log in the directory.

- Packer version = 1.3.0
- Host platform = Centos7
  
```

+ set +x

Creating AMI for QA-3-application

Template validated successfully.

amazon-ebs output will be in this color.


==> amazon-ebs: Force Deregister flag found, skipping prevalidating AMI Name

    amazon-ebs: Found Image ID: ami-a3373888

==> amazon-ebs: Using SSH Agent for existing key pair KEYPAIR

==> amazon-ebs: Launching a source AWS instance...

==> amazon-ebs: Adding tags to source instance


    amazon-ebs: Adding tag: "Stack": "app"

    amazon-ebs: Adding tag: "Name": "Packer Builder"

    amazon-ebs: Adding tag: "packer_mark": "757-1536696234129-497525"

    amazon-ebs: Adding tag: "Environment": "QA"

    amazon-ebs: Requesting spot instance 't2.medium' for: 0.50

==> amazon-ebs: No volumes to clean up, skipping

Build 'amazon-ebs' errored: unexpected EOF


==> Some builds didn't complete successfully and had errors:

--> amazon-ebs: unexpected EOF


==> Builds finished but no artifacts were created.

panic: runtime error: invalid memory address or nil pointer dereference

2018/09/11 20:03:57 packer: [signal SIGSEGV: segmentation violation code=0x1 addr=0x80 pc=0xd37c8f]

2018/09/11 20:03:57 packer: 

2018/09/11 20:03:57 packer: goroutine 25 [running]:

2018/09/11 20:03:57 packer: github.com/hashicorp/packer/builder/amazon/common.(*StepRunSpotInstance).Run(0xc000780000, 0x2819dc0, 0xc0001bf740, 0x2817bc0, 0xc00074a9c0, 0x0)

2018/09/11 20:03:57 packer: 	/Users/mmarsh/go/src/github.com/hashicorp/packer/builder/amazon/common/step_run_spot_instance.go:186 +0xa4f

2018/09/11 20:03:57 packer: github.com/hashicorp/packer/helper/multistep.(*BasicRunner).Run(0xc0001bf700, 0x2817bc0, 0xc00074a9c0)

2018/09/11 20:03:57 packer: 	/Users/mmarsh/go/src/github.com/hashicorp/packer/helper/multistep/basic_runner.go:72 +0x222

2018/09/11 20:03:57 packer: github.com/hashicorp/packer/builder/amazon/ebs.(*Builder).Run(0xc000474000, 0x281f640, 0xc000377fa0, 0x280d180, 0xc00050b8f0, 0x281a180, 0xc0004aa190, 0x4bca9c, 0x1e1b880, 0x1e1b880, ...)

2018/09/11 20:03:57 packer: 	/Users/mmarsh/go/src/github.com/hashicorp/packer/builder/amazon/ebs/builder.go:260 +0x1410

2018/09/11 20:03:57 packer: github.com/hashicorp/packer/packer/rpc.(*BuilderServer).Run(0xc0004a8080, 0x1, 0xc00032c030, 0x0, 0x0)

2018/09/11 20:03:57 packer: 	/Users/mmarsh/go/src/github.com/hashicorp/packer/packer/rpc/builder.go:94 +0x1cd

2018/09/11 20:03:57 packer: reflect.Value.call(0xc0004ac300, 0xc0004aa060, 0x13, 0x2544246, 0x4, 0xc000487f18, 0x3, 0x3, 0xc00007fd80, 0x413277, ...)

2018/09/11 20:03:57 packer: 	/usr/local/go/src/reflect/value.go:447 +0x449

2018/09/11 20:03:57 packer: reflect.Value.Call(0xc0004ac300, 0xc0004aa060, 0x13, 0xc000137718, 0x3, 0x3, 0x12a05f200, 0xc000137710, 0xc0001377b8)

2018/09/11 20:03:57 packer: 	/usr/local/go/src/reflect/value.go:308 +0xa4

2018/09/11 20:03:57 packer: net/rpc.(*service).call(0xc0004a4080, 0xc00049e0a0, 0xc0001d2118, 0xc0001d2120, 0xc0004a6280, 0xc0004a8300, 0x1e8b7e0, 0xc00032c02c, 0x18a, 0x1e1b880, ...)

2018/09/11 20:03:57 packer: 	/usr/local/go/src/net/rpc/server.go:384 +0x14e

2018/09/11 20:03:57 packer: created by net/rpc.(*Server).ServeCodec

2018/09/11 20:03:57 packer: 	/usr/local/go/src/net/rpc/server.go:481 +0x47e

2018/09/11 20:03:57 ui error: Build 'amazon-ebs' errored: unexpected EOF

2018/09/11 20:03:57 Builds completed. Waiting on interrupt barrier...

2018/09/11 20:03:57 machine readable: error-count []string{"1"}

2018/09/11 20:03:57 ui error: 

==> Some builds didn't complete successfully and had errors:

2018/09/11 20:03:57 machine readable: amazon-ebs,error []string{"unexpected EOF"}

2018/09/11 20:03:57 ui error: --> amazon-ebs: unexpected EOF

2018/09/11 20:03:57 ui: 

==> Builds finished but no artifacts were created.

2018/09/11 20:03:57 waiting for all plugin processes to complete...

2018/09/11 20:03:57 /ops/packer/1.3.0/packer: plugin process exited

2018/09/11 20:03:57 /ops/packer/1.3.0/packer: plugin process exited

2018/09/11 20:03:57 /ops/packer/1.3.0/packer: plugin process exited

2018/09/11 20:03:57 /ops/packer/1.3.0/packer: plugin process exited




!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!


Packer crashed! This is always indicative of a bug within Packer.

A crash log has been placed at "crash.log" relative to your current

working directory. It would be immensely helpful if you could please

report the crash with Packer[1] so that we can fix this.


[1]: https://github.com/hashicorp/packer/issues


!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

script returned exit code 1


```

--- a/builder/amazon/ebs/builder.go
+++ b/builder/amazon/ebs/builder.go
@@ -128,6 +128,7 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 			BlockDevices:                      b.config.BlockDevices,
 			BlockDurationMinutes:              b.config.BlockDurationMinutes,
 			Ctx:                               b.config.ctx,
+			Comm:                              &b.config.RunConfig.Comm,
 			Debug:                             b.config.PackerDebug,
 			EbsOptimized:                      b.config.EbsOptimized,
 			ExpectedRootDevice:                "ebs",

--- a/builder/amazon/ebssurrogate/builder.go
+++ b/builder/amazon/ebssurrogate/builder.go
@@ -142,6 +142,7 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 			BlockDevices:                      b.config.BlockDevices,
 			BlockDurationMinutes:              b.config.BlockDurationMinutes,
 			Ctx:                               b.config.ctx,
+			Comm:                              &b.config.RunConfig.Comm,
 			Debug:                             b.config.PackerDebug,
 			EbsOptimized:                      b.config.EbsOptimized,
 			ExpectedRootDevice:                "ebs",

--- a/builder/amazon/ebsvolume/builder.go
+++ b/builder/amazon/ebsvolume/builder.go
@@ -126,6 +126,7 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 			BlockDevices:                      b.config.launchBlockDevices,
 			BlockDurationMinutes:              b.config.BlockDurationMinutes,
 			Ctx:                               b.config.ctx,
+			Comm:                              &b.config.RunConfig.Comm,
 			Debug:                             b.config.PackerDebug,
 			EbsOptimized:                      b.config.EbsOptimized,
 			ExpectedRootDevice:                "ebs",

--- a/builder/amazon/instance/builder.go
+++ b/builder/amazon/instance/builder.go
@@ -212,6 +212,7 @@ func (b *Builder) Run(ui packer.Ui, hook packer.Hook, cache packer.Cache) (packe
 			BlockDevices:             b.config.BlockDevices,
 			BlockDurationMinutes:     b.config.BlockDurationMinutes,
 			Ctx:                      b.config.ctx,
+			Comm:                     &b.config.RunConfig.Comm,
 			Debug:                    b.config.PackerDebug,
 			EbsOptimized:             b.config.EbsOptimized,
 			IamInstanceProfile:       b.config.IamInstanceProfile,

